examples/rust/src/main.rs

```diff
@@ -0,0 +1,65 @@
+mod actions;
+mod rpc;
+mod state;
+
+use std::{env, fs};
+
+use anyhow::{anyhow, Result as AnyResult};
+use subxt::{
+    ext::{
+        sp_core::{crypto::Ss58Codec, sr25519::Pair, Pair as _},
+        sp_runtime::AccountId32,
+    },
+    tx::PairSigner,
+    Config, OnlineClient, PolkadotConfig,
+};
+
+#[subxt::subxt(runtime_metadata_url = "wss://ws.test.azero.dev:443")]
+pub mod aleph {}
+
+const TESTNET_WS: &str = "wss://ws.test.azero.dev:443";
+const HARDXORE_ADDRESS: &str = "5GErKuHmZ8ytupuZb78AJbHY9yoaDnKLdLUYKchYukhrsjVj";
+const BADGES: [&str; 5] = ["WARMUP", "XOR-0", "XOR-1", "XOR-2", "XOR-3"];
+
+const HAS_BADGE_SELECTOR: [u8; 4] = [0xfd, 0xdc, 0xef, 0x2b];
+const REGISTER_RANDOMNESS_SELECTOR: [u8; 4] = [0x0b, 0x81, 0x97, 0x41];
+const GET_RANDOMNESS_SELECTOR: [u8; 4] = [0x19, 0x4a, 0x46, 0xc8];
+const ATTEMPT_XOR_3_SELECTOR: [u8; 4] = [0xb7, 0x56, 0x5f, 0x19];
```

Comment:
TBH, I thought things like that are generated by `subxt` from the `metadata.json` of the contract.

Reply:
by cargo contract - yes; but I don't see anything in subxt for that

Reply:
There is a metadata file in `metadata/hardxore.json` could you leave a comment that these selectors come from this file :P?

Reply:
done

Reply:
We could also mark the methods with selectors in the contract itself:
```rust
#[ink(message, selector = 1)]
pub fn foo() -> { ... }
```
rather than using the compiled/autogenerated versions.

Reply:
that's true, but now it's after birds